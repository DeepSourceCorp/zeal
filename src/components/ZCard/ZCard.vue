<template>
  <div class="z-card">
    <div class="z-card__header" v-if="$slots.header">
      <slot name="header"></slot>
    </div>
    <div class="z-card__body" v-if="$slots.body">
      <slot name="body"></slot>
    </div>
    <div class="z-card__footer" v-if="$slots.footer">
      <slot name="footer"></slot>
    </div>
  </div>
</template>

<script lang="ts">
export default {
  name: 'ZCard'
}
</script>

<style lang="css" scoped>
.z-card {
  @apply text-vanilla-100 rounded-md overflow-hidden transition-all shadow;
}
</style>
