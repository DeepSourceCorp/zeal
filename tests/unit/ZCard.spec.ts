import ZCard from '../../src/components/ZCard'
import Vue from 'vue'

import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

describe('Card Component', () => {
  let mountFn: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue

  beforeEach(() => {
    mountFn = (options = {}) => {
      return mount(ZCard, {
        localVue,
        ...options
      })
    }
  })

  it('renders a Card component with Header, Body and Footer named slots', () => {
    const wrapper = mountFn({
      slots: {
        header:
          '<img src="https://www.kindpng.com/picc/m/4-46830_chris-evans-png-transparent-png.png" class="w-full block object-fill">',
        body:
          '<p>"DeepSource has made the code base <span class="text-juniper">much more stable and dependable</span>. It allowed us to identify many more areas for improvement."</p>',
        footer: '<p class="font-medium block">Chris Evans</p>'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
