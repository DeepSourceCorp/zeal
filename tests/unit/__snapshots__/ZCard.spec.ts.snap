// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders a Card component with Header, Body and Footer named slots 1`] = `
<div class="z-card text-vanilla-100 rounded-md overflow-hidden transition-all flex flex-col h-full">
  <div class="z-card__header"><img src="https://www.kindpng.com/picc/m/4-46830_chris-evans-png-transparent-png.png" class="w-full block object-fill"></div>
  <div class="z-card__body flex-1">
    <p>"DeepSource has made the code base <span class="text-juniper">much more stable and dependable</span>. It allowed us to identify many more areas for improvement."</p>
  </div>
  <div class="z-card__footer">
    <p class="font-medium block">Chris Evans</p>
  </div>
</div>
`;
